web/src/main/java/org/springframework/security/web/wellknown/package-info.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Classes which are responsible for the implementation of "well-known locations" RFC 8615.
+ *
+ * @author Evgeniy Cheban
+ * @see <a href="https://tools.ietf.org/html/rfc8615">RFC 8615</a>
+ */
+package org.springframework.security.web.wellknown;
```

Comment:
I hesitate to say that any well-known URI support would go into the same package - it seems more natural to align them with their business function (e.g. password, oidc, etc.).

One thing that's nice about this is that it makes it a bit easier to share APIs in a package-private way while they are maturing into something that can be exposed as public. It appears more likely that there will be shared support between various password management classes than there will between various `.well-known` endpoints.

Another nice thing about it is logging - if I want to trace all password request activity, then I can do so by changing just the one package logging level. It appears more likely that an application will want to bump up logging levels for all of `web.password` than all of `web.wellknown`.

TBH, I'm a little bit on the fence, but `web.password` is making more sense to me.

Reply:
Renamed to `web.password`.